src/resource/file.test.ts

```diff
@@ -376,339 +377,380 @@ describe("Write non-RDF data into a folder", () => {
+        const mockFetch = setMockOnFetch(jest.fn<typeof fetch>());
+
+        await saveFileInContainer("https://some.url", data, {
+          fetch: mockFetch,
+        });
+
+        const mockCall = mockFetch.mock.calls[0];
+        expect(mockCall[0]).toBe("https://some.url");
+        expect(mockCall[1]?.headers).toEqual({
+          "Content-Type":
+            mockBlob === data ? "binary" : "application/octet-stream",
+        });
+        expect(mockCall[1]?.body).toEqual(data);
+      });
+
+      it("should pass the suggested slug through", async () => {
+        const mockFetch = setMockOnFetch(jest.fn<typeof fetch>());
+
+        await saveFileInContainer("https://some.url", data, {
+          fetch: mockFetch,
+          slug: "someFileName",
+        });
+
+        const mockCall = mockFetch.mock.calls[0];
+        expect(mockCall[0]).toBe("https://some.url");
+        expect(mockCall[1]?.headers).toEqual({
+          "Content-Type":
+            mockBlob === data ? "binary" : "application/octet-stream",
+          Slug: "someFileName",
+        });
+        expect(mockCall[1]?.body).toEqual(data);
+      });
+
+      it("sets the correct Content Type on the returned file, if available", async () => {
+        const fetcher = jest.requireMock("../fetcher") as {
+          fetch: jest.Mocked<typeof window.fetch>;
+        };
+
+        fetcher.fetch = setMockOnFetch(fetcher.fetch);
+
+        const mockTextBlob = new Blob(["mock blob data"], {
+          type: "text/plain",
+        });
+        const savedFile = await saveFileInContainer(
+          "https://some.url",
+          mockTextBlob
+        );
+
+        expect(savedFile).toBeInstanceOf(Blob);
+        expect(savedFile!.internal_resourceInfo.contentType).toBe("text/plain");
+      });
+
+      it("sets the given Content Type on the returned file, if any was given", async () => {
+        const fetcher = jest.requireMock("../fetcher") as {
+          fetch: jest.Mocked<typeof window.fetch>;
+        };
+
+        fetcher.fetch = setMockOnFetch(fetcher.fetch);
+
+        const mockTextBlob = new Blob(["mock blob data"], {
+          type: "text/plain",
+        });
+        const savedFile = await saveFileInContainer(
+          "https://some.url",
+          mockTextBlob,
+          {
+            contentType: "text/csv",
+          }
+        );
+
+        expect(savedFile).toBeInstanceOf(Blob);
+        expect(savedFile!.internal_resourceInfo.contentType).toBe("text/csv");
+      });
+
+      it("defaults the Content Type to `application/octet-stream` if none is known", async () => {
+        const fetcher = jest.requireMock("../fetcher") as {
+          fetch: jest.Mocked<typeof window.fetch>;
+        };
+
+        fetcher.fetch = setMockOnFetch(fetcher.fetch);
+
+        const mockTextBlob = new Blob(["mock blob data"]);
+        const savedFile = await saveFileInContainer(
+          "https://some.url",
+          mockTextBlob
+        );
```

Comment:
Should the default `mockBlob` have no `type`, and this be replaced with:
```suggestion
        const savedFile = await saveFileInContainer(
          "https://some.url",
          data
        );
```
?

If so, we should have a separate test for the `instanceof` check, to make sure the generic typing works properly.